### PR TITLE
Stack queue animal shelter

### PR DESCRIPTION
**Pull request by Nicholas-Mercado**
[x ] Top-level README “Table of Contents” is updated
 - [x ] README for this challenge is complete
       - [x ] Summary, Description, Approach & Efficiency, Solution
       - [x ] Picture of whiteboard
       - [ x] Link to code
 - [ x] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
       - [x ] “Happy Path” - Expected outcome
       - [x ] Expected failure
       - [x ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: python/code_challenges/stack_queue_animal_shelter.py [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+
+class Dog(Animal):
+
+    def __init__(self, name=None, pref="dog"):
+        self.pref = pref
+
+class Cat(Animal):
+
+    def __init__(self, name=None, pref="cat"):
+        self.pref = pref
```
Comment (jstreifel-33) [inline python/code_challenges/stack_queue_animal_shelter.py:45]: I'm pretty sure I see what you're trying to do here, but it's worth noting that `__init__` from Dog and Cat are overriding the Animal `__init__`. Because of this, name isn't actually being assigned in your Dog and Cat objects. If you wanted to stick with this approach, you can use `super()` to access parent class methods that would normally be overridden, like this:
```Python
class Dog(Animal):
    def __init__(self, name=None, pref="dog"):
        super().__init__(name)
        self.pref = pref
```
